unit_tests/library/nearest.cpp

```diff
@@ -43,6 +43,67 @@ BOOST_AUTO_TEST_CASE(test_nearest_response)
     }
 }
 
+BOOST_AUTO_TEST_CASE(test_nearest_response_max_results)
+{
+    const auto args = get_args();
+    osrm::EngineConfig config;
+    config.max_results_nearest = 2;
+    // Default radius is unlimited
+    auto osrm = getOSRM(args.at(0), config);
+
+    using namespace osrm;
+
+    NearestParameters params;
+    params.number_of_results = 10;
+    params.coordinates.push_back(get_dummy_location());
+
+    json::Object result;
+    const auto rc = osrm.Nearest(params, result);
+    BOOST_REQUIRE(rc == Status::Ok);
+
+    const auto code = result.values.at("code").get<json::String>().value;
+    BOOST_CHECK_EQUAL(code, "Ok");
+
+    const auto &waypoints = result.values.at("waypoints").get<json::Array>().values;
+    BOOST_CHECK(waypoints.size() == 2); // make sure our results were capped at max_results_nearest
```

Comment:
As outlined above I think this should return the `TooBig` status code instead of silently capping results.